sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -104,6 +118,115 @@ class SentryOkHttpInterceptor(
         }
     }
 
+    // TODO: ignore exceptions of the type UnknownHostException
+
+    private fun captureEvent(request: Request, response: Response) {
+        // not possible to get a parameterized url, but we remove at least the
+        // query string and the fragment.
+        // url example: https://api.github.com/users/getsentry/repos/#fragment?query=query
+        // url will be: https://api.github.com/users/getsentry/repos/
+        // ideally we'd like a parameterized url: https://api.github.com/users/{user}/repos/
+        // but that's not possible
+        var requestUrl = request.url.toString()
+
+        val query = request.url.query
+        if (!query.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("?$query", "")
+        }
+
+        val urlFragment = request.url.fragment
+        if (!urlFragment.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("#$urlFragment", "")
+        }
+
+        if (!captureFailedRequests || !TracePropagationTargets.contain(failedRequestsTargets, requestUrl) || !containsStatusCode(response.code)) {
+            return
+        }
+
+        val mechanism = Mechanism().apply {
+            type = "SentryOkHttpInterceptor"
+        }
+        val exception = SentryHttpClientError("Event was captured because the request status code was ${response.code}")
+        val mechanismException = ExceptionMechanismException(mechanism, exception, Thread.currentThread(), true)
+        val event = SentryEvent(mechanismException)
+
+        val hint = Hint()
+        hint.set("request", request)
+        hint.set("response", response)
+
+        // TODO: remove after fields indexed
+        val tags = mutableMapOf<String, String>()
+        tags["status_code"] = response.code.toString()
+        tags["url"] = requestUrl
```

Comment:
This should be done on sentry/snuba.
Snuba already has the contexts indexed, but we'd need to add it to its own list of searchable fields.
https://github.com/getsentry/sentry/blob/master/src/sentry/snuba/events.py

Example for [http.url](https://github.com/getsentry/sentry/blob/master/src/sentry/snuba/events.py#L229-L235)

Example with a context field https://github.com/getsentry/sentry/blob/master/src/sentry/snuba/events.py#L236-L242

Still checking about the `event.request.url`.

Reply:
https://github.com/getsentry/sentry/pull/39913